### PR TITLE
Autotools integration

### PR DESCRIPTION
**Pull request by tapia**

### CODE REVIEW

File: integrations/autotools/building_with_autotools.rst [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+      settings = "os", "compiler", "build_type", "arch"
+      requires = "poco/1.9.4"
+      default_options = {"poco:shared": True, "openssl:shared": True}
+
+      def imports(self):
+         self.copy("*.dll", dst="bin", src="bin")
+         self.copy("*.dylib*", dst="bin", src="lib")
+
+      def build(self):
+         autotools = AutoToolsBuildEnvironment(self)
```
Comment (memsharded) [inline integrations/autotools/building_with_autotools.rst:26]: Legacy integration, needs to be migrated.

File: integrations/autotools/creating_packages.rst [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+       def build(self):
+           with tools.chdir("src"):
+               atools = Autotools(self)
+               atools.make()
+
+       def package(self):
+           self.copy("*app", dst="bin", keep_path=False)
+           self.copy("*app.exe", dst="bin", keep_path=False)
+
+       def deploy(self):
```
Comment (memsharded) [inline integrations/autotools/creating_packages.rst:148]: Better not use ``deploy()`` method at the moment, it is still to be discussed. Use ``VirtualRunEnv`` to run things

File: integrations/autotools/building_with_autotools.rst [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+      def imports(self):
+        self.copy("*.dll", dst="bin", src="bin")
+        self.copy("*.dylib*", dst="bin", src="lib")
+
+      def build(self):
+         env_build = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+         env_build.configure()
+         env_build.make()
+
+Constructor
```
Comment (memsharded) [inline integrations/autotools/building_with_autotools.rst:87]: This is reference-like docs, it shouldn't go here, but redirect to the ``reference`` section.

File: integrations/autotools/building_with_autotools.rst (hunk 1 above)
Comment (tapia) [reply integrations/autotools/building_with_autotools.rst:26]: I know, this is just a draft PR yet

File: integrations/autotools/creating_packages.rst (hunk 2 above)
Comment (tapia) [reply integrations/autotools/creating_packages.rst]: Done

**Pull request closed without merging**